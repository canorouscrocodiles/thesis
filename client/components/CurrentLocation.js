import React, { Component } from 'react'
import { connect } from 'react-redux'

class CurrentLocation extends Component {
  constructor (props) {
    super(props)
    this.state = {}
  }

  render () {
    return (
      <div>
        <h4>Current Location</h4>
        <p>South of Market St</p>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return { currentLocation: state.currentLocation }
}

export default connect(mapStateToProps)(CurrentLocation)
