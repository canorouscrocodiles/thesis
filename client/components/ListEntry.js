import React from 'react'
import { Link } from 'react-router-dom'

export default (props) => (
  <div className='list-entry'>
    <p><Link className='link post-title' to={`/question/${props.question.id}`}>{props.question.message}</Link></p>
    <p className='question-location'>{props.question.location}</p>
    <p>{props.question.username} - {props.question.timestamp}</p>
    <p>{props.question.category}</p>
  </div>
)
