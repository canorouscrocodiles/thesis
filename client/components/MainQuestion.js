import React, { Component } from 'react'
import { connect } from 'react-redux'

class MainQuestion extends Component {
  constructor (props) {
    super(props)
    this.state = {}
  }

  render () {
    return (
      <div>
        <p>Keith - 22s ago</p>
        <p>Is Chipotle out of guac?!</p>
        <p>Hack Reactor</p>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return { questions: state.questions }
}

export default connect(mapStateToProps)(MainQuestion)
