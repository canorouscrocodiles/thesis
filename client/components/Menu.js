import React from 'react'

export default (props) => (
  <div>
    <p>Home</p>
    <p>Search</p>
    <p>Notifications</p>
    <p>Me</p>
  </div>
)
