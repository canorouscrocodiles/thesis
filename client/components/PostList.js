import React, { Component } from 'react'
import { connect } from 'react-redux'
import { sortQuestions, changeOption, changeValue } from '../actions/questions'
import ListEntry from './ListEntry'
import UnreadNotification from './UnreadNotification'
import { Dimmer, Divider, Dropdown, Header, Icon, Image, Loader, Segment } from 'semantic-ui-react'

class PostList extends Component {
  constructor (props) {
    super(props)
    this.state = {
      value: [],
      sortOptions: [
        { key: 'New', value: 'New', text: 'New' },
        { key: 'Trending', value: 'Trending', text: 'Trending' },
        { key: 'Distance', value: 'Distance', text: 'Distance' },
        { key: 'Old', value: 'Old', text: 'Old' }
      ],
      defaultSort: '',
      defaultCategories: []
    }

    this.handleOptionChange = this.handleOptionChange.bind(this)
    this.handleCategoryChange = this.handleCategoryChange.bind(this)
  }

  componentDidMount () {
    this.setState({ defaultSort: this.props.option, defaultCategories: this.props.categories })
  }

  handleOptionChange (event, data) {
    console.log(data.value)
    this.props.changeOption(data.value)

    // let selectedCategories = this.props.value.map(x => x.value)
    this.props.sortQuestions(data.value, this.props.value)
  }

  handleCategoryChange (event, data) {
    console.log('You\'ve selected:', data.value)
    // this.props.changeValue(data.value)

    console.log('sortBy ', this.props.option)
    this.props.sortQuestions(this.props.option, data.value)
  }

  locationLoaded () {
    let district
    let city
    let location = this.props.currentLocation.name.address_components
    if (location.length > 1) {
      district = location[2].long_name
      city = location[3].long_name
    } else {
      district = 'United States'
      city = 'The Earth'
    }
    return (
      <h3>
        <Icon name='marker' color='grey' size='big' />Questions around { `${district}, ${city}` }
      </h3>
    )
  }

  renderMessage () {
    if (!this.props.currentLocation.name) {
      return (
        <h3><Loader indeterminate active inline /> Finding your location...</h3>
      )
    }
    return this.locationLoaded()
  }

  renderUnread () {
    if (this.props.unread) {
      return <UnreadNotification />
    }
  }

  renderPostList () {
    if (this.props.questions.data.length === 0 && this.props.currentLocation.location) {
      return (
        <Segment raised textAlign='center'>
          <Header as='h3'>There are no questions around you.</Header>
          <Header as='h3'>ðŸ˜ž</Header>
          <Header as='h3'>Post a question you want answered!</Header>
        </Segment>
      )
    } else if (this.props.questions.data.length === 0) {
      return (
        <Segment raised>
          <Dimmer active inverted>
            <Loader size='massive' inline='centered'>Loading Questions</Loader>
          </Dimmer>
          <Image src='https://react.semantic-ui.com/assets/images/wireframe/paragraph.png' />
          <Divider />
          <Image src='https://react.semantic-ui.com/assets/images/wireframe/paragraph.png' />
        </Segment>
      )
    } else {
      return (
        <div id='postlist' className='post-list'>
          {this.props.questions.data.map(question => question.active ? <ListEntry key={question.id} question={question} /> : null)}
        </div>
      )
    }
  }

  render () {
    return (
      <div className='questions'>
        <Segment basic>
          <Segment textAlign='center' raised>
            {this.renderMessage()}
            <Dropdown
              placeholder='Sort'
              selection
              defaultValue={this.state.defaultSort}
              options={this.state.sortOptions}
              onChange={this.handleOptionChange}
            />
            <Dropdown
              placeholder='Filter categories'
              multiple
              selection
              defaultValue={this.state.defaultCategories}
              options={this.props.categoryOptions}
              onChange={this.handleCategoryChange}
            />
          </Segment>
          <Divider hidden />
          {this.renderUnread()}
          {this.renderPostList()}
        </Segment>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    questions: state.questions,
    currentLocation: state.currentLocation,
    sortBy: state.questions.sortBy,
    categories: state.questions.categories,
    categoryOptions: state.questions.categoryOptions,
    unread: state.questions.unread,
    value: state.questions.value,
    sortOptions: state.questions.sortOptions,
    option: state.questions.option
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    sortQuestions: (sortBy, categories) => dispatch(sortQuestions(sortBy, categories)),
    changeOption: (option) => dispatch(changeOption(option)),
    changeValue: (value) => dispatch(changeValue(value))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(PostList)
