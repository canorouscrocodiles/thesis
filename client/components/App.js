import React, { Component } from 'react'
import GMap from './GMap'
import { connect } from 'react-redux'
import CurrentLocation from './CurrentLocation'
import Menu from './Menu'
import PostList from './PostList'

class App extends Component {
  constructor (props) {
    super(props)
    this.state = {}
  }

  render () {
    return (
      <div>
        <h2>OnPoint ðŸ‘‡</h2>
        <Menu />
        <GMap />
        <CurrentLocation />
        <PostList />
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return { user: state.user }
}

export default connect(mapStateToProps)(App)
