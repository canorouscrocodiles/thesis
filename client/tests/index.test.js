import React from 'react'
import App from '../components/App'
import CurrentLocation from '../components/CurrentLocation'
import ListEntry from '../components/ListEntry'
import MainQuestion from '../components/MainQuestion'
import Menu from '../components/Menu'
import PostList from '../components/PostList'
import renderer from 'react-test-renderer'

it('renders App component to DOM', () => {
  const app = renderer.create(<App />
  ).toJSON()
  expect(app).toMatchSnapshot()
})

it('renders PostList component to DOM', () => {
  const app = renderer.create(<PostList />
  ).toJSON()
  expect(app).toMatchSnapshot()
})

it('renders ListEntry component to DOM', () => {
  const app = renderer.create(<ListEntry />
  ).toJSON()
  expect(app).toMatchSnapshot()
})

it('renders CurrentLocation component to DOM', () => {
  const app = renderer.create(<CurrentLocation />
  ).toJSON()
  expect(app).toMatchSnapshot()
})

it('renders Menu component to DOM', () => {
  const app = renderer.create(<Menu />
  ).toJSON()
  expect(app).toMatchSnapshot()
})

it('renders MainQuestion component to DOM', () => {
  const app = renderer.create(<MainQuestion />
  ).toJSON()
  expect(app).toMatchSnapshot()
})
