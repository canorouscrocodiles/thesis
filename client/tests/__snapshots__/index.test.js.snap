// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component to DOM 1`] = `
<div>
  <h2>
    OnPoint ðŸ‘‡
  </h2>
  <div>
    <p>
      Home
    </p>
    <p>
      Search
    </p>
    <p>
      Notifications
    </p>
    <p>
      Me
    </p>
  </div>
  <div>
    <h4>
      Current Location
    </h4>
    <p>
      South of Market St
    </p>
  </div>
  <div>
    <h3>
      PostList
    </h3>
    <div>
      <p>
        Keith - 10m Ago
      </p>
      <p>
        What does Tri Tip even mean?
      </p>
    </div>
    <div>
      <p>
        Keith - 10m Ago
      </p>
      <p>
        What does Tri Tip even mean?
      </p>
    </div>
    <div>
      <p>
        Keith - 10m Ago
      </p>
      <p>
        What does Tri Tip even mean?
      </p>
    </div>
    <div>
      <p>
        Keith - 10m Ago
      </p>
      <p>
        What does Tri Tip even mean?
      </p>
    </div>
  </div>
</div>
`;

exports[`renders CurrentLocation component to DOM 1`] = `
<div>
  <h4>
    Current Location
  </h4>
  <p>
    South of Market St
  </p>
</div>
`;

exports[`renders ListEntry component to DOM 1`] = `
<div>
  <p>
    Keith - 10m Ago
  </p>
  <p>
    What does Tri Tip even mean?
  </p>
</div>
`;

exports[`renders MainQuestion component to DOM 1`] = `
<div>
  <p>
    Keith - 22s ago
  </p>
  <p>
    Is Chipotle out of guac?!
  </p>
  <p>
    Hack Reactor
  </p>
</div>
`;

exports[`renders Menu component to DOM 1`] = `
<div>
  <p>
    Home
  </p>
  <p>
    Search
  </p>
  <p>
    Notifications
  </p>
  <p>
    Me
  </p>
</div>
`;

exports[`renders PostList component to DOM 1`] = `
<div>
  <h3>
    PostList
  </h3>
  <div>
    <p>
      Keith - 10m Ago
    </p>
    <p>
      What does Tri Tip even mean?
    </p>
  </div>
  <div>
    <p>
      Keith - 10m Ago
    </p>
    <p>
      What does Tri Tip even mean?
    </p>
  </div>
  <div>
    <p>
      Keith - 10m Ago
    </p>
    <p>
      What does Tri Tip even mean?
    </p>
  </div>
  <div>
    <p>
      Keith - 10m Ago
    </p>
    <p>
      What does Tri Tip even mean?
    </p>
  </div>
</div>
`;
